app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
```

Comment:
We have a new helper component `form-button-component` (helper of the main form), and since it's related to form operations, it will need those 2 injects.